### ISSUE TITLE
Rubocop slowness

### ISSUE THREAD

**Issue by DawidJanczak** [github/super-linter#424]
Hello!

I'm not sure this is something that can be easily fixed/acted upon, but linting Ruby is incredibly slow on large codebases. It takes a little over a minute to run it inside my docker container on 1043 files, but the same number of files takes super linter over 22 minutes to finish. If I understand this comment https://github.com/github/super-linter/issues/234#issuecomment-647225064 correctly rubocop is started anew for each file and that's probably the cause of slowness. Would it be possible to instead run the command once?

**Comment by jwiebalk** [comment]
@DawidJanczak are you using `VALIDATE_ALL_CODEBASE=true`? Otherwise it should only be linting the files that have been changed in the commit/PR.

The linters all run on a single file each time since there should only be a few files changed. We could look at running against entire directories at once when doing the validate all codebase though.

**Comment by DawidJanczak** [comment]
@jwiebalk thanks for the response! No, I don't, I have the following variables set:
```
      env:
        LINTER_RULES_PATH: .
        RUBY_CONFIG_FILE: .rubocop.yml
        VALIDATE_RUBY: true
```
That being said I do see `- Validating ALL files in code base...` in the logs.

This happens on a PR introducing the linter so I figured it's because of that perhaps, but then I opened another test PR on top of this one changing only one file and still all files are being checked.
